### PR TITLE
test: creating delete markers w/ null ver

### PR DESCRIPTION
**Pull request by electrachong**
AWS has this description in its [REST API](http://docs.aws.amazon.com/AmazonS3/latest/API/RESTObjectDELETE.html) which is different from our own implementation: The DELETE operation removes the null version (if there is one) of an object and inserts a delete marker, which becomes the current version of the object. If there isn't a null version, Amazon S3 does not remove any objects.

These tests were run against AWS and verify despite what the description implies, that our implementation is compatible's with AWS's. The DELETE operation only removes the null version if versioning is suspended, not if it is enabled.

### CODE REVIEW

**Comment by electrachong** [comment]
@LaurenSpiegel updated the tests

**Comment by electrachong** [comment]
@ironman-machine

**Comment by electrachong** [comment]
@ironman-machine

File: tests/functional/aws-node-sdk/test/versioning/objectDelete.js [hunk 1]
```diff
@@ -4,11 +4,119 @@ import async from 'async';
...
+describe('delete marker creation in bucket with null version', () => {
+    withV4(sigCfg => {
+        const bucketUtil = new BucketUtility('default', sigCfg);
+        const s3 = bucketUtil.s3;
+        const nullVersionBody = 'nullversionbody';
+
+        beforeEach(done => {
+            s3.createBucket({ Bucket: bucket }, err => {
+                if (err) {
+                    done(err);
```
Comment (nicolas2bert) [inline tests/functional/aws-node-sdk/test/versioning/objectDelete.js]: return done(err) ?

File: tests/functional/aws-node-sdk/test/versioning/objectDelete.js [hunk 2]
```diff
@@ -4,11 +4,119 @@ import async from 'async';
...
+                } // put null object
+                s3.putObject({
```
Comment (nicolas2bert) [inline tests/functional/aws-node-sdk/test/versioning/objectDelete.js]: return s3.putObject({ ?

File: tests/functional/aws-node-sdk/test/versioning/objectDelete.js (hunk 1 above)
Comment (electrachong) [inline tests/functional/aws-node-sdk/test/versioning/objectDelete.js]: thanks 👍

File: tests/functional/aws-node-sdk/test/versioning/objectDelete.js (hunk 2 above)
Comment (electrachong) [inline tests/functional/aws-node-sdk/test/versioning/objectDelete.js]: 👍

**Comment by electrachong** [comment]
end to end passed

**Pull request merged**